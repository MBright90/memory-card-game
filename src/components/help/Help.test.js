import { appContext } from '@app'
import Help from '@components/helpLink'
import { fireEvent, render, screen } from '@testing-library/react'

describe('Help', () => {
  const setIsHelpActive = jest.fn()

  it('renders a title which says "How to play', () => {
    const context = { setIsHelpActive }

    render(<Help />, {
      wrapper: ({ children }) => (
        <appContext.Provider value={context}>{children}</appContext.Provider>
      )
    })

    const title = screen.getByText(/How to Play/i)
    expect(title).toBeInTheDocument()
  })

  it('renders two information paragraphs', () => {
    const context = { setIsHelpActive }

    render(<Help />, {
      wrapper: ({ children }) => (
        <appContext.Provider value={context}>{children}</appContext.Provider>
      )
    })

    const infoParagraphs = screen.getAllByRole('contentinfo')
    expect(infoParagraphs).toHaveLength(2)
  })

  it('renders a link to play the game', () => {
    const context = { setIsHelpActive }

    render(<Help />, {
      wrapper: ({ children }) => (
        <appContext.Provider value={context}>{children}</appContext.Provider>
      )
    })

    const link = screen.getByRole('link')
    expect(link).toBeInTheDocument()
  })

  it('calls handlePlayClick when the link is clicked', () => {
    const context = { setIsHelpActive }
    const handlePlayClick = jest.fn()

    render(<Help />, {
      wrapper: ({ children }) => (
        <appContext.Provider value={context}>{children}</appContext.Provider>
      )
    })

    const link = screen.getByRole('link')

    fireEvent.click(link)
    expect(handlePlayClick).toHaveBeenCalled()

    fireEvent.keyDown(link, { key: 'Enter' })
    expect(handlePlayClick).toHaveBeenCalled()
  })

  it('calls handlePlayClick when the link is highlighted and pressed "Enter" on', () => {
    const context = { setIsHelpActive }
    const handlePlayClick = jest.fn()

    render(<Help />, {
      wrapper: ({ children }) => (
        <appContext.Provider value={context}>{children}</appContext.Provider>
      )
    })

    const link = screen.getByRole('link')

    fireEvent.keyDown(link, { key: 'Enter' })
    expect(handlePlayClick).toHaveBeenCalled()
  })
})
