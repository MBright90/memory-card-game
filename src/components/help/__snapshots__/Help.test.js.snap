// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help renders to match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h1
      class="helpTitle"
    >
      How to Play
    </h1>
    <p
      class="infoPara"
      role="contentinfo"
    >
      You will see 
      <span
        class="emphasized"
      >
        twelve
      </span>
       character cards. You must choose one of the cards. Try to remember which card you chose, as each time you choose a card, the cards will be shuffled into a random order. Choose another character card that is different from the one you chose before. You
       
      <span
        class="emphasized"
      >
        cannot
      </span>
       choose the same card twice.
    </p>
    <p
      class="infoPara"
      role="contentinfo"
    >
      Repeat these steps until you have chosen all 12 character cards without repeating any choices. If the you successfully choose all 12 different cards without making any duplicate choices, you win the game. If you make a duplicate choice, your score is reset to zero and you must start again from the beginning.
    </p>
    <p
      class="infoPara"
      role="contentinfo"
    >
      Most importantly...
      <span
        class="emphasized"
      >
        Have fun!
      </span>
    </p>
    <div
      class="playButton"
      data-testid="play-link"
      role="link"
      tabindex="-3"
    >
      <a
        href="#/"
      >
        Play
      </a>
    </div>
  </div>
</div>
`;
